ResNet_constructor.ipynb

```diff
@@ -0,0 +1,1393 @@
+{
```

Comment:
<p>1. outputs = outputs.to(device) -- do you really need this? afair, the outputs should already be on the device</p><p>2. <code></p><p>&nbsp;&nbsp;&nbsp;&nbsp;for images, labels in trainloader:</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;images, labels = images.to(device), labels.to(device)</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;outputs = model(images)</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;outputs = outputs.to(device)</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;_, predicted = torch.max(outputs.data, 1)</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;total_train += labels.size(0)</p><p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;correct_train += (predicted == labels).sum().item()</p><p></code> </p><p>-- this code is repeated twice, consider putting it in a separate function</p><p> </p><p>3. <code>metrics_frame = metrics_frame.append(pd.DataFrame.from_dict(metrics,orient='index').T)</p><p>&nbsp;&nbsp;metrics_frame.to_csv(metrics_frame_file,index=False)</code></p><p>-- this is inefficient. If you want to write some lines to a file, read about <code>file.write</code>  and <code>file.writelines</code>  </p><p> </p><p>4. <code>for images, labels in trainloader:</code></p><p> -- do you really need to iterate over all the training data and all the test data each epoch?</p><p> </p><p>5. Don't you think that splitting this huge sheet into functions will be better?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bestetc/May/pull/1/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Reply:
<ol><li>agreed</li><li>rewrited soon</li><li>diff between file.writelines and dataframe.append+dataframe.to_csv more that 10 times, but it less than 1 ms in this case. file.writelines more efficient.</li><li>this point most uneffective in this notebook. effective decision release in ResNetD_constructor. Add it to this notebook good idea.</li><li>agreed</li></ol>